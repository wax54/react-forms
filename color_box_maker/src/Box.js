import React from "react";
import './Box.css';


const Box = ({height, width, color, deleteBox, id}) => {
    const boxStyle = { height: `${height}px`, width: `${width}px`, backgroundColor: color }
    return (
        <>
            <div className="Box" style={boxStyle} data-testid="BoxElement"></div>
            <button onClick={() => deleteBox(id)}>X</button>
        </>
    );
}

export default Box;
