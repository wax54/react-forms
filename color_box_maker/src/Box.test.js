import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import Box from './Box';

it("Doesn't Blow up", () => {
    render(<Box />);
});

it('should match previous snapshot', function () {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});



