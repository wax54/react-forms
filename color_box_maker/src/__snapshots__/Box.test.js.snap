// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="BoxElement"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
